pumps/sqs.go

```diff
@@ -0,0 +1,177 @@
+package pumps
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/TykTechnologies/tyk-pump/analytics"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/aws/aws-sdk-go-v2/service/sqs/types"
+	"github.com/kr/pretty"
+	"github.com/mitchellh/mapstructure"
+	"github.com/oklog/ulid/v2"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+type SQSSendMessageBatchAPI interface {
+	GetQueueUrl(ctx context.Context,
+		params *sqs.GetQueueUrlInput,
+		optFns ...func(*sqs.Options)) (*sqs.GetQueueUrlOutput, error)
+
+	SendMessageBatch(ctx context.Context,
+		params *sqs.SendMessageBatchInput,
+		optFns ...func(*sqs.Options)) (*sqs.SendMessageBatchOutput, error)
+}
+
+type SQSPump struct {
+	SQSClient   SQSSendMessageBatchAPI
+	SQSQueueURL *string
+	SQSConf     *SQSConf
+	log         *logrus.Entry
+	CommonPumpConfig
+}
+
+var SQSPrefix = "sqs-pump"
+var SQSDefaultENV = PUMPS_ENV_PREFIX + "_SQS" + PUMPS_ENV_META_PREFIX
+
+// @PumpConf SQS
+type SQSConf struct {
+	EnvPrefix                        string `mapstructure:"meta_env_prefix"`
+	QueueName                        string `mapstructure:"aws_queue_name"`
+	AWSRegion                        string `mapstructure:"aws_region"`
+	AWSSecret                        string `mapstructure:"aws_secret"`
+	AWSKey                           string `mapstructure:"aws_key"`
+	AWSEndpoint                      string `mapstructure:"aws_endpoint"`
+	AWSDelaySeconds                  int32  `mapstructure:"aws_delay_seconds"`
+	AWSMessageGroupID                string `mapstructure:"aws_message_group_id"`
+	AWSSQSBatchLimit                 int    `mapstructure:"aws_sqs_batch_limit"`
+	AWSMessageIDDeduplicationEnabled bool   `mapstructure:"aws_message_id_deduplication_enabled"`
```

Comment:
Could you add an explanatory comment to each field?

Reply:
@mativm02 Hi
Thanks for your review and notes you mentioned. 
I've committed new changes according to them.